frontend/benefit/applicant/src/components/applications/forms/application/step2/useApplicationFormStep2.ts

```diff
@@ -35,9 +42,62 @@ type ExtendedComponentProps = {
   getDefaultSelectValue: (fieldName: keyof Application) => OptionType;
 };
 
+const ssnValidationTable = [
+  '0',
+  '1',
+  '2',
+  '3',
+  '4',
+  '5',
+  '6',
+  '7',
+  '8',
+  '9',
+  'A',
+  'B',
+  'C',
+  'D',
+  'E',
+  'F',
+  'H',
+  'J',
+  'K',
+  'L',
+  'M',
+  'N',
+  'P',
+  'R',
+  'S',
+  'T',
+  'U',
+  'V',
+  'W',
+  'X',
+  'Y',
+];
+
+const validateSsn = (number: number, checkCharacter: string): boolean =>
+  checkCharacter === ssnValidationTable[number % 31];
+
+export const mustBeSsn = (value = ''): boolean => {
+  if (!value) return true;
+
+  const ssnRe = SSN_REGEX;
+
+  if (!ssnRe.test(value)) return false;
+
+  const groups = ssnRe.exec(value) ?? [];
+  const number = parseInt(`${groups[1]}${groups[3]}`, 10);
+  const checkCharacter = groups[4];
+
+  if (!validateSsn(number, checkCharacter)) return false;
+
+  return true;
+};
+
 const useApplicationFormStep2 = (
   application: Application
-): ExtendedComponentProps => {
+): UseApplicationFormStep2Props => {
```

Comment:
What is the point of renaming to "Use..."?

Reply:
Nothing really 😅 
I saw some other hooks following this naming convention; so I thought, why not this one as well?